full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,9 @@
+import { products, type Product } from './data';
+
+export function readCatalog(): Promise<Product[]> {
+  return Promise.resolve(products);
```

Comment:
So, replace this with a fetch call, plus the response.ok check and response.json() call.